console-subscriber/src/lib.rs

```diff
@@ -50,7 +51,18 @@ struct WakerVisitor {
     op: Option<WakeOp>,
 }
 
-struct Watch(mpsc::Sender<Result<proto::tasks::TaskUpdate, tonic::Status>>);
+struct Watch<T>(mpsc::Sender<Result<T, tonic::Status>>);
+
+enum WatchKind {
+    Task(Watch<proto::tasks::TaskUpdate>),
```

Comment:
nit: i think this should be called "task*s*", rather than "task", "task" suggests that it is a watch on a  _single_ task